source/expectations/equal.tpb

```diff
@@ -0,0 +1,178 @@
+        l_actual   ut_data_value_blob := treat(a_actual as ut_data_value_blob);
+      begin
+        l_result := equal_with_nulls((dbms_lob.compare( l_expected.value, l_actual.value) = 0), a_actual);
+      end;
+    elsif self.expected is of (ut_data_value_boolean) and a_actual is of (ut_data_value_boolean) then
+      declare
+        l_expected ut_data_value_boolean := treat(self.expected as ut_data_value_boolean);
+        l_actual   ut_data_value_boolean := treat(a_actual as ut_data_value_boolean);
+      begin
+        l_result := equal_with_nulls((l_expected.value = l_actual.value), a_actual);
+      end;
+    elsif self.expected is of (ut_data_value_clob) and a_actual is of (ut_data_value_clob) then
+      declare
+        l_expected ut_data_value_clob := treat(self.expected as ut_data_value_clob);
+        l_actual   ut_data_value_clob := treat(a_actual as ut_data_value_clob);
+      begin
+        l_result := equal_with_nulls((dbms_lob.compare( l_expected.value, l_actual.value) = 0), a_actual);
+      end;
+    elsif self.expected is of (ut_data_value_date) and a_actual is of (ut_data_value_date) then
+      declare
+        l_expected ut_data_value_date := treat(self.expected as ut_data_value_date);
+        l_actual   ut_data_value_date := treat(a_actual as ut_data_value_date);
+      begin
+        l_result := equal_with_nulls((l_expected.value = l_actual.value), a_actual);
+      end;
+    elsif self.expected is of (ut_data_value_number) and a_actual is of (ut_data_value_number) then
+      declare
+        l_expected ut_data_value_number := treat(self.expected as ut_data_value_number);
+        l_actual   ut_data_value_number := treat(a_actual as ut_data_value_number);
+      begin
+        l_result := equal_with_nulls((l_expected.value = l_actual.value), a_actual);
+      end;
+    elsif self.expected is of (ut_data_value_refcursor) and a_actual is of (ut_data_value_refcursor) then
+      declare
+        l_expected ut_data_value_refcursor := treat(self.expected as ut_data_value_refcursor);
+        l_actual   ut_data_value_refcursor := treat(a_actual as ut_data_value_refcursor);
+      begin
+        if l_expected.value is not null and l_actual.value is not null then
+          --fetch 1M rows max
+          dbms_xmlgen.setMaxRows(l_expected.value, 1000000);
+          dbms_xmlgen.setMaxRows(l_actual.value, 1000000);
+          --TODO - set formatting of date/timestamp/number for XML
+          l_result := dbms_lob.compare( dbms_xmlgen.getxml(l_expected.value), dbms_xmlgen.getxml(l_actual.value) ) = 0;
+          --TODO - clear formatting of date/timestamp/number for XML
+        else
+          l_result := equal_with_nulls( null, a_actual);
+        end if;
+      end;
+    elsif self.expected is of (ut_data_value_timestamp) and a_actual is of (ut_data_value_timestamp) then
+      declare
+        l_expected ut_data_value_timestamp := treat(self.expected as ut_data_value_timestamp);
```

Comment:
i see what youve done here to avoid the case statement that was causing iissues last week..  i like that you've changed the whole lot to match..  +1